tfe/resource_tfe_policy_set.go

```diff
@@ -0,0 +1,300 @@
+
+			for _, policyID := range newPolicies.List() {
+				options.PolicyIDs = append(options.PolicyIDs, policyID.(string))
+			}
+
+			log.Printf("[DEBUG] Add policies to policy set: %s", d.Id())
+			err := tfeClient.PolicySets.AddPolicies(ctx, d.Id(), options)
+			if err != nil {
+				return fmt.Errorf("Error adding policies to policy set %s: %v", d.Id(), err)
+			}
+		}
+
+		// Then remove all the old policies.
+		if oldPolicies.Len() > 0 {
+			options := tfe.PolicySetPoliciesRelationshipOptions{}
+
+			for _, policyID := range oldPolicies.List() {
+				options.PolicyIDs = append(options.PolicyIDs, policyID.(string))
+			}
+
+			log.Printf("[DEBUG] Remove policies from policy set: %s", d.Id())
+			err := tfeClient.PolicySets.RemovePolicies(ctx, d.Id(), options)
+			if err != nil {
+				return fmt.Errorf("Error removing policies from policy set %s: %v", d.Id(), err)
+			}
+		}
+	}
+
+	if !global && d.HasChange("workspaces") {
+		old, new := d.GetChange("workspaces")
+		oldWorkspaces := old.(*schema.Set).Difference(new.(*schema.Set))
+		newWorkspaces := new.(*schema.Set).Difference(old.(*schema.Set))
+
+		// Now add the new workspaces.
+		if newWorkspaces.Len() > 0 {
+			options := tfe.PolicySetWorkspacesRelationshipOptions{}
+
+			for _, workspacesID := range newWorkspaces.List() {
+				options.WorkspaceIDs = append(options.WorkspaceIDs, workspacesID.(string))
+			}
+
+			log.Printf("[DEBUG] Add workspaces to policy set: %s", d.Id())
+			err := tfeClient.PolicySets.AttachToWorkspaces(ctx, d.Id(), options)
+			if err != nil {
+				return fmt.Errorf("Error adding workspaces to policy set %s: %v", d.Id(), err)
+			}
+		}
+
+		// Then remove all the old workspaces.
+		if oldWorkspaces.Len() > 0 {
+			options := tfe.PolicySetWorkspacesRelationshipOptions{}
+
+			for _, workspaceID := range oldWorkspaces.List() {
+				options.WorkspaceIDs = append(options.WorkspaceIDs, workspaceID.(string))
+			}
+
+			log.Printf("[DEBUG] Remove workspaces from policy set: %s", d.Id())
+			err := tfeClient.PolicySets.DetachFromWorkspaces(ctx, d.Id(), options)
+			if err != nil {
+				return fmt.Errorf("Error removing workspaces from policy set %s: %v", d.Id(), err)
+			}
+		}
+	}
+
+	return resourceTFEPolicySetRead(d, meta)
+}
+
+func resourceTFEPolicySetDelete(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	log.Printf("[DEBUG] Delete policy set: %s", d.Id())
+	err := tfeClient.PolicySets.Delete(ctx, d.Id())
+	if err != nil {
+		if err == tfe.ErrResourceNotFound {
+			return nil
+		}
+		return fmt.Errorf("Error deleting policy set %s: %v", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceTFEPolicySetImporter(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	s := strings.SplitN(d.Id(), "/", 2)
+	if len(s) != 2 {
+		return nil, fmt.Errorf("invalid policy set import format: %s", d.Id())
+	}
+
+	// Set the fields that are part of the import ID.
+	d.Set("organization", s[0])
```

Comment:
I don't think this is needed here as a `PolicySet` struct contains a relation to the organization and so it can be retrieved from there. This is inline with the comment I left about reading the org name back in the read function and it would make this import function not needed anymore. Instead we can just use the policy set ID with the `schema.ImportStatePassthrough` importer.